pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -202,41 +241,35 @@ func (c *EtcdCertSignerController) sync(ctx context.Context, syncCtx factory.Syn
 	return updateErr
 }
 
-func (c *EtcdCertSignerController) syncAllMasterCertificates(ctx context.Context, recorder events.Recorder) error {
-	// TODO(thomas): it is of utmost importance to keep the existing signer certs for now
-	// when we just create a new signer cert, the new revision does not allow the peer to join the existing two-node
-	// cluster based on the old CA. Any newly rotated additions will come for free through the signerCaBundle and will work out of the box.
-	// Unfortunately, we can't make use of the certrotation.RotatedSigningCASecret here because it would immediately try to rotate the existing signer.
-	signerCaPair, err := tlshelpers.ReadConfigSignerCert(ctx, c.secretClient)
-	if err != nil {
-		return err
-	}
-
-	c.reportExpirationMetric(signerCaPair, "signer-ca")
-
-	// TODO(thomas): we need to transition that new signer as a replacement for the above - today we only bundle it
-	newSignerCaPair, _, err := c.certConfig.signerCert.EnsureSigningCertKeyPair(ctx)
+func (c *EtcdCertSignerController) syncAllMasterCertificates(
+	ctx context.Context, recorder events.Recorder, forceSkipRollout bool, lastRotationRevision int32, currentRevision int32) error {
+	signerCaPair, _, err := c.certConfig.signerCert.EnsureSigningCertKeyPair(ctx)
 	if err != nil {
 		return fmt.Errorf("error on ensuring etcd-signer cert: %w", err)
 	}
+	c.reportExpirationMetric(signerCaPair, "signer-ca")
 
-	metricsSignerCaPair, err := tlshelpers.ReadConfigMetricsSignerCert(ctx, c.secretClient)
+	metricsSignerCaPair, _, err := c.certConfig.metricsSignerCert.EnsureSigningCertKeyPair(ctx)
 	if err != nil {
-		return err
+		return fmt.Errorf("error on ensuring metrics-signer cert: %w", err)
 	}
-
 	c.reportExpirationMetric(metricsSignerCaPair, "metrics-signer-ca")
 
-	// TODO(thomas): we need to transition that new signer as a replacement for the above - today we only bundle it
-	newMetricsSignerCaPair, _, err := c.certConfig.metricsSignerCert.EnsureSigningCertKeyPair(ctx)
+	signerBundle, metricsSignerBundle, rolloutTriggered, err := c.ensureBundles(ctx, recorder, signerCaPair, metricsSignerCaPair, currentRevision)
 	if err != nil {
-		return fmt.Errorf("error on ensuring metrics-signer cert: %w", err)
+		return fmt.Errorf("error on ensuring bundles: %w", err)
 	}
 
-	signerBundle, metricsSignerBundle, err := c.ensureBundles(ctx, recorder,
-		[]*crypto.CA{signerCaPair, newSignerCaPair}, []*crypto.CA{metricsSignerCaPair, newMetricsSignerCaPair})
-	if err != nil {
-		return fmt.Errorf("error on ensuring bundles: %w", err)
+	if !forceSkipRollout {
+		if rolloutTriggered {
```

Comment:
Just confirming that we'll skip here as we first trigger the rollout:
`skipping EtcdCertSignerController leaf cert generation as revision rollout has been triggered`
and then the controller will loop for a couple mins on L213 until the revision stabilizes:
`skipping EtcdCertSignerController can't get current revision. Err=revision rollout in progress, can't establish current revision`

Basically pause during rollouts.

Reply:
correct